### PR TITLE
*: clean up the opt package

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Some code enhancement

### What is changed and how it works?

Clean up the unnecessary code in opt package.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by HunDunDM** [comment]
`Cluster` is an `Interface` defined in `opt`, which can be `RaftCluster` or `MockCluster`. Using `RuleManager` to `FitRegion` is the behavior defined on the `Cluster` interface, not the logic in `RaftCluster`.

**Comment by rleungx** [comment]
> `Cluster` is an interface defined in `opt`, which can be `RaftCluster` or `MockCluster`. `CacheCluster` should be based on `Cluster` interface instead of `RaftCluster`.
> 
> Using `RuleManager` to `FitRegion` is the behavior defined on the `Cluster` interface like `IsRegionHealthy`.

IMO, CacheCluster is not very useful in mock cluster. As for FitRegion, we have already had RuleManager in cluster interface. Maybe remove it make the package more clear?

File: server/config/config.go [hunk 1]
```diff
@@ -1184,6 +1184,10 @@ type StoreLabel struct {
 	Value string `toml:"value" json:"value"`
 }
 
+// RejectLeader is the label property type that suggests a store should not
+// have any region leaders.
+const RejectLeader = "reject-leader"
```
Comment (HunDunDM) [inline server/config/config.go:1189]: how about `server/core/store.go`?

**Review by HunDunDM — COMMENTED** [review]
rest LGTM

File: server/config/config.go (hunk 1 above)
Comment (rleungx) [reply server/config/config.go:1189]: I think it can be regarded as a kind of config to some degree.

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**